### PR TITLE
establish plugin loading dock

### PR DESCRIPTION
**Pull request by arlimus**

### CODE REVIEW

**Comment by chef-supermarket** [comment]
Hi. I am an automated pull request bot named Curry. There are commits in this pull request whose authors are not yet authorized to contribute to Chef Software, Inc. projects or are using a non-GitHub verified email address. To become authorized to contribute, you will need to sign the Contributor License Agreement (CLA) as an individual or on behalf of your company. [You can read more on Chef's blog.](https://www.getchef.com/blog/2014/06/23/changes-to-the-contributor-license-agreement-process)

## GitHub Users Who Are Not Authorized To Contribute

The following GitHub users do not appear to have signed a CLA:

* @arlimus

[Please sign the CLA here.](https://supermarket.chef.io/become-a-contributor)

File: lib/inspec/plugins.rb [hunk 1]
```diff
@@ -6,4 +6,37 @@ module Inspec
...
+               .map { |x| File.dirname(x) }
+               .map { |x| Dir[File.join(x, 'lib', 'inspec-*.rb')] }
+               .flatten
+      # map paths to names
+      @registry = Hash[@paths.map { |x|
+        [File.basename(x, '.rb'), x]
+      }]
+    end
+
+    def list
```
Comment (srenatus) [inline lib/inspec/plugins.rb]: Nitpicking, but how about `extend Forwardable` and `def_delegator :@registry, :keys, :list`?

Comment (arlimus) [inline lib/inspec/plugins.rb]: great catch! not yet used to forwardable, but your suggestion simplifies code really well :) thank you for that!

**Comment by chris-rock** [comment]
@arlimus so cool to get this started!

**Pull request merged**